opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -689,27 +698,24 @@ namespace Opm {
             // Set the well primary variables based on the value of well solutions
             initPrimaryVariablesEvaluation();
 
-            std::vector< Scalar > B_avg(numComponents(), Scalar() );
-            computeAverageFormationFactor(B_avg);
-
```

Comment:
This potentially changes result, because it is not updated anymore for each iteration.  Do we have any document about what the correct way should be?

Reply:
The B_avg does not change much during simulation. With this change it is only updated every report step i.e. when the well is created, I will be very surprised if this has any noticeable effect on the results.

Reply:
Sure. I understand it, while it probably depends on the individual case setup. 

There are some other places also changes the iteration, we will evaluate it. 

Most ideally, if this PR only output the well potentials for the cases that request the output without changing any running results, it will be much easier to review and test. 

If the iteration changes, it just adds some testing burden for the cases we have concerns with.

Reply:
> There are some other places also changes the iteration, we will evaluate it.

If you send me an email with the name of the models that changes results due to this I could take a look myself.